client/src/backoffice/AddressManagementPage.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { FC, useState, useEffect } from 'react';
+import classNames from 'classnames';
+
+/* Libraries */
+import ReactModal from 'react-modal';
+import { Formik, FormikActions, Form, Field, FieldProps, ErrorMessage } from 'formik';
+
+/* Helpers */
+import { GasPriceSchema } from '../lib/schemas';
+import { getSigners, setSigner, AdminAddress } from '../api';
+import { convertToGWEI, convertFromGWEI, convertToETH, reduceAddress } from '../lib/helpers';
+
+/* Components */
+import { SubmitButton } from '../components/SubmitButton';
+import { Loading } from '../components/Loading';
+
+/* Assets */
+import edit from '../images/edit.svg';
+
+type GasPriceFormValues = {
+  gasPrice: string;
+};
+
+ReactModal.setAppElement('#root');
+
+const AddressManagementPage: FC = () => {
+  const [isFetchingAddresses, setIsFetchingAddresses] = useState<null | boolean>(null);
+  const [addresses, setAddresses] = useState<null | AdminAddress[]>(null);
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [selectedAddress, setSelectedAddress] = useState<null | AdminAddress>(null);
+
+  useEffect(() => {
+    fetchSigners();
+  }, []);
+
+  const fetchSigners = () => {
+    setIsFetchingAddresses(true);
+    setAddresses(null);
+
+    getSigners()
+      .then(addresses => {
+        if (!addresses) return;
+        setAddresses(addresses);
+      })
+      .catch(error => console.error(error))
+      .finally(() => setIsFetchingAddresses(false));
+  }
+
+  const handleFormSubmit = async (
+    values: GasPriceFormValues,
+    actions: FormikActions<GasPriceFormValues>
+  ) => {
+    if (!selectedAddress) return;
+    try {
+      actions.setStatus(null);
+      actions.setSubmitting(true);
+
+      const gasPriceInWEI = convertFromGWEI(values.gasPrice);
+      await setSigner(selectedAddress.id, gasPriceInWEI);
+      fetchSigners();
```

Comment:
Do you want to wait for `fetchSigners` to resolve and then close the modal?

Reply:
Yes, I wan't to show the table with the new value I just modified.
If not, I could be making a PUT request and at the same time the GET, and the backend might not make the update in time.

Reply:
Great, but we are not awaiting for `fetchSigners` to resolve or maybe I'm missing something.
And I guess that's ok since we are displaying the `isFetchingAddresses` flag in the table and not in the modal, so we need to close the modal immediately.

Reply:
My bad, I confused `setSigner` and `fetchSigners`. Let me recap:
I want to await `setSigner` so that I'm sure that the change has been made in the database.
Now, regarding waiting `fetchSigners`, as you said I'm setting `isFetchingAddresses` as true and showing a preloader in the table, so I don't need to wait for it. So I don't see the point in waiting for that request.

Reply:
Oh now we're on the same page! Thanks for the clarification.